exporter/awscloudwatchlogsexporter/exporter.go

```diff
@@ -18,111 +18,131 @@ import (
+		logger:           params.Logger,
+		retryCount:       *awsConfig.MaxRetries,
+		collectorID:      collectorIdentifier.String(),
+	}
+	return logsExporter, nil
 }
 
-func (e *exporter) PushLogs(ctx context.Context, ld pdata.Logs) (err error) {
-	// TODO(jbd): Relax this once CW Logs support ingest
-	// without sequence tokens.
-	e.seqTokenMu.Lock()
-	defer e.seqTokenMu.Unlock()
+func newCwLogsExporter(config config.Exporter, params component.ExporterCreateSettings) (component.LogsExporter, error) {
+	logsExporter, err := newCwLogsPusher(config, params)
+	if err != nil {
+		return nil, err
+	}
+	expConfig := config.(*Config)
+	return exporterhelper.NewLogsExporter(
+		config,
+		params,
+		logsExporter.ConsumeLogs,
+		exporterhelper.WithQueue(expConfig.enforcedQueueSettings()),
+		exporterhelper.WithRetry(expConfig.RetrySettings),
+	)
+
+}
 
+func (e *exporter) ConsumeLogs(ctx context.Context, ld pdata.Logs) error {
+	exp := e.config.(*Config)
+	cwLogsPusher := e.getLogPusher(exp.LogGroupName, exp.LogStreamName)
 	logEvents, _ := logsToCWLogs(e.logger, ld)
 	if len(logEvents) == 0 {
 		return nil
 	}
 
-	e.logger.Debug("Putting log events", zap.Int("num_of_events", len(logEvents)))
-	input := &cloudwatchlogs.PutLogEventsInput{
-		LogGroupName:  aws.String(e.config.LogGroupName),
-		LogStreamName: aws.String(e.config.LogStreamName),
-		LogEvents:     logEvents,
+	for _, logEvent := range logEvents {
+		logEvent := &cwlogs.Event{
+			InputLogEvent: logEvent,
+			GeneratedTime: time.Now(),
+		}
+		e.logger.Debug("Adding log event", zap.Any("event", logEvent))
+		err := cwLogsPusher.AddLogEntry(logEvent)
+		if err != nil {
+			e.logger.Error("Failed ", zap.Int("num_of_events", len(logEvents)))
+		}
 	}
-	if e.seqToken != "" {
-		input.SequenceToken = aws.String(e.seqToken)
-	} else {
-		e.logger.Debug("Putting log events without a sequence token")
+	e.logger.Debug("Log events are successfully put")
+	flushErr := cwLogsPusher.ForceFlush()
+	if flushErr != nil {
+		e.logger.Error("Error force flushing logs. Skipping to next logPusher.", zap.Error(flushErr))
+		return flushErr
 	}
+	return nil
+}
 
-	out, err := e.client.PutLogEvents(input)
-	if err != nil {
-		return err
-	}
-	if info := out.RejectedLogEventsInfo; info != nil {
-		return fmt.Errorf("log event rejected: %s", info.String())
+func (e *exporter) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+func (e *exporter) Shutdown(ctx context.Context) error {
+	if e.pusher != nil {
+		e.pusher.ForceFlush()
 	}
-	e.logger.Debug("Log events are successfully put")
+	return nil
+}
 
-	e.seqToken = *out.NextSequenceToken
+func (e *exporter) Start(ctx context.Context, host component.Host) error {
 	return nil
 }
 
+func (e *exporter) getLogPusher(logGroup, logStream string) cwlogs.Pusher {
+	if e.pusher == nil {
+		e.pusher = cwlogs.NewPusher(aws.String(logGroup), aws.String(logStream), e.retryCount, *e.svcStructuredLog, e.logger)
```

Comment:
All of this information is available in `newCwLogsPusher()` where an `exporter` is created, so I would suggest doing this there and accessing the `e.pusher` field directly in `ConsumeLogs()` rather than through this method.